unblob/pool.py

```diff
@@ -1,13 +1,23 @@
 import abc
 import multiprocessing as mp
+import os
 import queue
+import signal
 import sys
 import threading
 from multiprocessing.queues import JoinableQueue
 from typing import Any, Callable, Union
 
+from structlog import get_logger
+
 from .logging import multiprocessing_breakpoint
 
+logger = get_logger()
+
+
+class ShutDownRequired(Exception):
```

Comment:
I'd inherit from `BaseException` so that it escapes `try ... except Exception` blocks like `KeyboardInterrupt` does.